spec/models/contribution_spec.rb

```diff
@@ -95,71 +95,23 @@
-    it 'returns true for a valid authorship hash' do
-      expect(described_class.featured_valid?(authorship)).to be true
-    end
-    it 'returns false for an authorship hash without "featured"' do
-      authorship['featured'] = nil
-      expect(described_class.featured_valid?(authorship)).to be false
-    end
-  end
-
-  describe '.status_valid?' do
-    it 'returns true for a valid authorship hash' do
-      expect(described_class.status_valid?(authorship)).to be true
-    end
-    it 'returns true for an upper case field value' do
-      authorship['status'].upcase!
-      expect(described_class.status_valid?(authorship)).to be true
-    end
-    it 'returns false for an authorship hash without "status"' do
-      authorship.delete 'status'
-      expect(described_class.status_valid?(authorship)).to be false
-    end
-    it 'returns false for an authorship hash with a nil "status"' do
-      authorship['status'] = nil
-      expect(described_class.status_valid?(authorship)).to be false
-    end
-    it 'returns false for an authorship hash with an invalid "status"' do
-      authorship['status'] = 'invalid value'
-      expect(described_class.status_valid?(authorship)).to be false
-    end
-  end
-
-  describe '.visibility_valid?' do
-    it 'returns true for a valid authorship hash' do
-      expect(described_class.visibility_valid?(authorship)).to be true
-    end
-    it 'returns true for an upper case field value' do
-      authorship['visibility'].upcase!
-      expect(described_class.visibility_valid?(authorship)).to be true
-    end
-    it 'returns false for an authorship hash without "visibility"' do
-      authorship.delete 'visibility'
-      expect(described_class.visibility_valid?(authorship)).to be false
-    end
-    it 'returns false for an authorship hash with a nil "visibility"' do
-      authorship['visibility'] = nil
-      expect(described_class.visibility_valid?(authorship)).to be false
-    end
-    it 'returns false for an authorship hash with an invalid "visibility"' do
-      authorship['visibility'] = 'invalid value'
-      expect(described_class.visibility_valid?(authorship)).to be false
+    context 'for invalid authorship hashes' do
+      it 'returns false for bad values' do
+        expect(described_class.valid_fields?(authorship.merge(status: 'XYZ'))).to be false
+        expect(described_class.valid_fields?(authorship.merge(visibility: 'transparent'))).to be false
+      end
+      it 'returns false without "featured"' do
+        authorship['featured'] = nil
+        expect(described_class.valid_fields?(authorship)).to be false
+      end
+      it 'returns false without "status"' do
+        authorship['status'] = nil
+        expect(described_class.valid_fields?(authorship)).to be false
+      end
+      it 'returns false without "visibility"' do
+        authorship['visibility'] = nil
+        expect(described_class.valid_fields?(authorship)).to be false
+      end
     end
```

Comment:
What if `visibility` etc. are an empty String?  IIRC, all these attributes are validated against enums now, so is it safe to assume that an empty String is false?  Why is `featured` ommitted from the first `bad values` block?

Reply:
Right, empty string is not an enumerated value, so it won't validate via AR. 

`featured` defaults to `false` when nil (reasonably enough), so at the model level, it is not actually required.  Some parts of the API require it, but not the model.

Reply:
:ok_hand: